state/unit.go

```diff
@@ -1447,7 +1448,22 @@ func (u *Unit) charm() (CharmRefFull, error) {
 	if cURL == nil {
 		return nil, errors.Errorf("missing charm URL for %q", u.Name())
 	}
-	ch, err := u.st.Charm(*cURL)
+
+	var ch CharmRefFull
+	err := retry.Call(retry.CallArgs{
+		Attempts: 8,
+		Delay:    10 * time.Millisecond,
```

Comment:
I suggest moving this back to 50ms, or up the number of attempts. 80ms is not a long time at all. I'd probably want to see 1 second being the upper limit.

Reply:
Oh oops I set this to 10 for testing, it's meant to be 25. But I will up to 50 and set the attempts to 20.

Reply:
1s seems like as long time, but as we say, this is all dying soon, so we can err on the side of cautious